### PR TITLE
ios-webkit-debug-proxy: init at 1.9.0

### PR DESCRIPTION
**Pull request by abustany**
###### Description of changes

Add package to connect to iOS Safari/webviews from Linux.

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [x] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/mobile/ios-webkit-debug-proxy/default.nix
Comment (Janik-Haag) [inline pkgs/development/mobile/ios-webkit-debug-proxy/default.nix:15]: We want to get rid of this pattern, for context see: https://github.com/NixOS/nixpkgs/pull/209325

**Review by Janik-Haag — CHANGES_REQUESTED** [review]

**Comment by Janik-Haag** [comment]
Result of `nixpkgs-review pr 232212` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package blacklisted:</summary>
  <ul>
    <li>nixos-install-tools</li>
  </ul>
</details>
<details>
  <summary>2 packages built:</summary>
  <ul>
    <li>ios-webkit-debug-proxy</li>
    <li>ios-webkit-debug-proxy.dev</li>
  </ul>
</details>

**Comment by Janik-Haag** [comment]
The binary works fine, and the PR generally looks quite in my opinion except for the two things mentioned above.

File: pkgs/development/mobile/ios-webkit-debug-proxy/default.nix
Comment (abustany) [reply pkgs/development/mobile/ios-webkit-debug-proxy/default.nix]: hmm very good point, that copy pasted that without thinking much, but as far as I know the code should work on any UNIX platform (assuming the dependencies also do). I removed this line.

File: pkgs/development/mobile/ios-webkit-debug-proxy/default.nix
Comment (abustany) [reply pkgs/development/mobile/ios-webkit-debug-proxy/default.nix]: woops sorry, I wasn't aware :grimacing: I moved the `with lib` to the `meta` section only, is that acceptable?

Comment (Janik-Haag) [reply pkgs/development/mobile/ios-webkit-debug-proxy/default.nix]: Jup that's fine :D

**Comment by Janik-Haag** [comment]
No problem, feel free to ping me in the future :D

**Comment by Janik-Haag** [comment]
Result of `nixpkgs-review pr 232212` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package blacklisted:</summary>
  <ul>
    <li>nixos-install-tools</li>
  </ul>
</details>
<details>
  <summary>2 packages built:</summary>
  <ul>
    <li>ios-webkit-debug-proxy</li>
    <li>ios-webkit-debug-proxy.dev</li>
  </ul>
</details>

**Review by Janik-Haag — APPROVED** [review]
LGTM

**Comment by abustany** [comment]
Hi, is there anything I need to do to get the PR merged?

**Comment by Janik-Haag** [comment]
I don't have commit right's to this repo so I cant merge the PR. Sometimes a person just randomly come's by and merges, there is also a discourse discussion](https://discourse.nixos.org/t/prs-ready-for-review/3032/2268) you can post the pr at, which I just did and you can also ask in the matrix space.

**Comment by Janik-Haag** [comment]
Sorry but there really isn't much more you can do except for waiting and maybe if you know someone  with commit privileges pinging them.

File: pkgs/development/mobile/ios-webkit-debug-proxy/default.nix
Comment (SuperSandro2000) [inline pkgs/development/mobile/ios-webkit-debug-proxy/default.nix:41]:
```suggestion
  preConfigure = ''
    NOCONFIGURE=1 ./autogen.sh
  '';
```

File: pkgs/development/mobile/ios-webkit-debug-proxy/default.nix
Comment (SuperSandro2000) [inline pkgs/development/mobile/ios-webkit-debug-proxy/default.nix:46]:
```suggestion
    description = "A DevTools proxy (Chrome Remote Debugging Protocol) for iOS devices (Safari Remote Web Inspector).";
```

File: nixos/doc/manual/release-notes/rl-2305.section.md
Comment (SuperSandro2000) [inline nixos/doc/manual/release-notes/rl-2305.section.md:163]: This should be moved to 23.11 or be dropped since the package is probably not that note worthy

Comment (abustany) [reply nixos/doc/manual/release-notes/rl-2305.section.md]: I removed this change.

File: pkgs/development/mobile/ios-webkit-debug-proxy/default.nix
Comment (abustany) [reply pkgs/development/mobile/ios-webkit-debug-proxy/default.nix]: fixed

File: pkgs/development/mobile/ios-webkit-debug-proxy/default.nix
Comment (abustany) [reply pkgs/development/mobile/ios-webkit-debug-proxy/default.nix]: fixed

**Comment by abustany** [comment]
@SuperSandro2000 hi :) I addressed your review comments, any chance we can get this one merged? Thanks!

**Pull request merged**
